aquasec/resource_group.go

```diff
@@ -52,23 +54,21 @@ func resourceGroupCreate(d *schema.ResourceData, m interface{}) error {
 	}
 	d.Set("group_id", group.Id)
 
-	err = resourceGroupRead(d, m)
-	if err != nil {
-		i := fmt.Sprintf("%v", group.Id)
-		d.SetId(i)
-	} else {
-		return err
-	}
-
-	return nil
+	d.SetId(fmt.Sprintf("%v", group.Id))
+	return resourceGroupRead(d, m)
 }
 
 func resourceGroupRead(d *schema.ResourceData, m interface{}) error {
 	ac := m.(*client.Client)
 
-	id := d.Get("group_id").(int)
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return nil
+	}
 	r, err := ac.GetGroup(id)
 	if err == nil {
+		d.Set("name", r.Name)
+		d.Set("group_id", r.Id)
```

Comment:
duplicate set for group_id and id

Reply:
what do you mean by duplicate set? group_id is part of the state and id is something terraform uses to track the resource